homeassistant/components/proximity/coordinator.py

```diff
@@ -0,0 +1,246 @@
+"""Data update coordinator for the Proximity integration."""
+
+import logging
+from typing import Any
+
+from homeassistant.const import (
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    CONF_DEVICES,
+    CONF_UNIT_OF_MEASUREMENT,
+    CONF_ZONE,
+    UnitOfLength,
+)
+from homeassistant.core import HomeAssistant, State
+from homeassistant.helpers.event import async_track_state_change
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util.location import distance
+from homeassistant.util.unit_conversion import DistanceConverter
+
+from .const import (
+    ATTR_DIR_OF_TRAVEL,
+    ATTR_DIST_TO,
+    ATTR_NEAREST,
+    CONF_IGNORED_ZONES,
+    CONF_TOLERANCE,
+    DEFAULT_DIR_OF_TRAVEL,
+    DEFAULT_DIST_TO_ZONE,
+    DEFAULT_NEAREST,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ProximityDataUpdateCoordinator(
+    DataUpdateCoordinator[dict[str, str | int | float]]
+):
+    """Proximity data update coordinator."""
+
+    def __init__(
+        self, hass: HomeAssistant, friendly_name: str, config: ConfigType
+    ) -> None:
+        """Initialize the Proximity coordinator."""
+        self.ignored_zones: list[str] = config[CONF_IGNORED_ZONES]
+        self.proximity_devices: list[str] = config[CONF_DEVICES]
+        self.tolerance: int = config[CONF_TOLERANCE]
+        self.proximity_zone: str = config[CONF_ZONE]
+        self.unit_of_measurement: str = config.get(
+            CONF_UNIT_OF_MEASUREMENT, hass.config.units.length_unit
+        )
+        self.friendly_name = friendly_name
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=friendly_name,
+            update_interval=None,
```

Comment:
Given this is not polling, and registering its own callbacks for tracking state changes, i this really meeting the spirit of a `DataUpdateCoordinator`? It seems like it may just be its own object for tracking state so i'm wondering if perhaps its not getting value from subclassing this, or if its not a good idea to subclass.

Reply:
i considered to go with `DataUpdateCoordinator` because it already brings everything needed, so registered entities get notified and values updates as soon as the data update on the coordinator was successful. this is also to be seen as prework for adding the sensor entities.